ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -414,6 +400,22 @@ public static KsqlRestApplication buildApplication(
     );
   }
 
+  private static String getKafkaClusterId(final KsqlConfig ksqlConfig) {
+
+    try (AdminClient client = AdminClient.create(ksqlConfig.getKsqlAdminClientConfigProps())) {
+
+      return client.describeCluster().clusterId().get();
+
+    } catch (final UnsupportedVersionException e) {
+      throw new KsqlException(
+          "The kafka brokers are incompatible with. "
+          + "KSQL requires broker versions >= 0.10.1.x"
+      );
+    } catch (final Exception e) {
+      throw new KsqlException("Failed to get Kafka cluster information", e);
```

Comment:
Why have a catch-all here?

Reply:
Whoops I thought I deleted this comment. Having a human-friendly error message here makes sense.